### PR TITLE
Add ESLint to the project and update some npm packages

### PR DESCRIPTION
**Pull request by kbberker**
The aim of these changes is to add linting to the project and update some npm packages.

The linting should help keep code consistent, stick to best practices and help people to contribute. There were security warnings with some of the npm packages due to their versions.

The main thing that has been added is the `eslintrc.js` file, which has all the ESLint setup. It's using the recommended React rules, as well as the airbnb rules for the javascript.

The reason the change is so big is that it also includes fixes for the linting. 

This is looking like quite a big change so might need to be broken up.

### CODE REVIEW

File: src/study.js [hunk 1]
```diff
@@ -265,27 +272,11 @@ export default class Study extends Component {
...
-        className="study__introduction__image"
-        style={{ backgroundImage: `url(${image})` }}
-      />
-    );
-  }
-}
-
 Study.propTypes = {
-  study: PropTypes.object.isRequired,
+  study: PropTypes.shape.isRequired,
```
Comment (jamesdoc) [inline src/study.js:276]: This is a nice feature that I didn't know about. Should the particular shape be referenced?

File: package.json [hunk 2]
```diff
@@ -3,15 +3,16 @@
...
     "moment": "2.19.3",
     "node-sass-chokidar": "1.2.2",
-    "npm": "^6.2.0",
-    "npm-run-all": "^4.1.2",
-    "prop-types": "^15.6.0",
-    "react": "^16.4.1",
-    "react-dom": "^16.4.1",
-    "react-router": "^4.2.0",
-    "react-router-dom": "^4.2.2",
+    "npm": "^6.14.9",
```
Comment (jamesdoc) [inline package.json:9]: I'm looking at this… trying to work out _why_ `npm` is a dependancy here… Have you got any ideas?!

File: package.json [hunk 3]
```diff
@@ -3,15 +3,16 @@
...
+    "npm-run-all": "^4.1.5",
```
Comment (jamesdoc) [inline package.json:10]: I'm going to rip this out. It's not needed.

File: src/App.js [hunk 4]
```diff
@@ -1,5 +1,7 @@
-import React, { Component, createContext } from "react";
+/* eslint-disable react/jsx-props-no-spreading */
+import React, { Component } from "react";
 import { BrowserRouter as Router, Route, Redirect } from "react-router-dom";
+import ApiEndpoint from "./ApiEndpoint";
```
Comment (jamesdoc) [inline src/App.js:4]: This is a *much* nicer way of doing things!

File: src/study.js (hunk 1 above)
Comment (kbberker) [reply src/study.js:276]: Yes you're right, this hasn't been completed properly. I don't think I had worked out what the shape should be yet.

Comment (kbberker) [reply src/study.js:276]: I've added the shape now from observing a console log.

**Comment by kbberker** [comment]
I have fixed all but 8 eslint issues! The ones that are left are a little bit more pernickety or I'm not quite sure how it works or both.
Those changes are in the `linting` branch. If you're happy I could merge `linting` into `kbb` and we can the merge that in?

**Pull request merged**